inc/inventorycomputerinventory.class.php

```diff
@@ -412,7 +412,18 @@ function sendCriteria($p_DEVICEID, $arrayinventory) {
          }
          $inputdb['rules_id'] = $data['_ruleid'];
          $inputdb['method'] = 'inventory';
-         $pfIgnoredimportdevice->add($inputdb);
+         $inputdb['plugin_fusioninventory_agents_id'] = $_SESSION['plugin_fusioninventory_agents_id'];
+
+         // if existing ignored device, update it
+         if ($found = $pfIgnoredimportdevice->find("`plugin_fusioninventory_agents_id` =
+                                                    '".$inputdb['plugin_fusioninventory_agents_id']."'
+                                                    AND `serial`='".$inputdb['serial']."'")) {
```

Comment:
use find to get only 1 item, not need get all, so the query will be faster with a `limit 1`